diff --git a/includes/ckeditor.utils.js b/includes/ckeditor.utils.js
index ce22cd4..efda28f 100644
--- a/includes/ckeditor.utils.js
+++ b/includes/ckeditor.utils.js
@@ -163,13 +163,18 @@ if (typeof window.CKEDITOR_BASEPATH === 'undefined') {
     }
 
     if (run_filter && ($("#" + textarea_id).val().length > 0) && typeof(ckeditor_obj.input_formats[ckeditor_obj.elements[textarea_id]]) != 'undefined' && ((ckeditor_obj.input_formats[ckeditor_obj.elements[textarea_id]]['ss'] == 1 && typeof(Drupal.settings.ckeditor.autostart) != 'undefined' && typeof(Drupal.settings.ckeditor.autostart[textarea_id]) != 'undefined') || ckeditor_obj.input_formats[ckeditor_obj.elements[textarea_id]]['ss'] == 2)) {
+      var input_format = ckeditor_obj.textarea_default_format[textarea_id];
+      if (!input_format) {
+        var sel_format = $("#" + textarea_id.substr(0, textarea_id.lastIndexOf("-")) + "-format--2");
+        input_format = sel_format.val();
+      }
       $.ajax({
         type: 'POST',
         url: Drupal.settings.ckeditor.xss_url,
         async: false,
         data: {
           text: $('#' + textarea_id).val(),
-          input_format: ckeditor_obj.textarea_default_format[textarea_id],
+          input_format: input_format,
           token: Drupal.settings.ckeditor.ajaxToken
         },
         success: function(text){
